src/file/rename.h

```diff
@@ -7,175 +7,127 @@
 #include "common/types.h"
 #include "common/warning.h"
 #include "dentry.h"
+#include "file/maps.h"
 #include "push_file_message.h"
 
-typedef struct {
-    u64 pid_tgid;
-    u64 start_ktime_ns;
-    void *vfsmount;                     // vfsmount of the source AND the destination.
-    void *source_dentry;                // dentry of the file we are
-                                        // moving. By the time we exit
-                                        // the name + parent will be the target name + parent
-    void *source_parent_dentry;         // The directory we are moving from
-    file_ownership_t overwr_owner;      // The owner of the overwritten file
-    file_info_t overwr;                 // Metadata of the overwritten file
-} incomplete_rename_t;
-
-typedef struct {
-    u64 pid_tgid;
-    char name[NAME_MAX+1];
+// used for events with flexible sizes (i.e., exec*) so it can send
+// extra data. Used in conjuction with a map such that it does not use
+// the stack size limit.
+typedef struct
+{
+    char name[NAME_MAX + 1];
 } rename_name_t;
 
-struct bpf_map_def SEC("maps/incomplete_renames") incomplete_renames = {
-    .type = BPF_MAP_TYPE_LRU_HASH,
-    .key_size = sizeof(u64),
-    .value_size = sizeof(incomplete_rename_t),
-    .max_entries = 512,
-    .pinning = 0,
-    .namespace = "",
-};
-
+// A per cpu buffer that can hold more data than allowed in the
+// stack. Used to collect data of variable length such as a string.
 struct bpf_map_def SEC("maps/rename_names") rename_names = {
```

Comment:
question (non-blocking): How does this map get updated?

Feel like I'm blind but I'm not seeing it 😞 .